pandas-stubs/core/strings.pyi

```diff
@@ -65,6 +65,10 @@ class StringMethods(NoNewAttributesMixin, Generic[T, _TS, _TM]):
         self, pat: str = ..., *, n: int = ..., expand: Literal[True], regex: bool = ...
     ) -> _TS: ...
     @overload
+    def split(
+        self, pat: str = ..., *, n: int = ..., expand: Literal[False], regex: bool = ...
+    ) -> Series[list[str]]: ...
```

Comment:
Adding this overload this way will cause `Index.str.split()` to return the wrong result.

So you have to figure out a way to support both `Index.str.split()` and `Series.str.split()` within this PYI file.